pageserver/src/bin/pageserver.rs

```diff
@@ -337,9 +348,13 @@ fn start_pageserver(conf: &'static PageServerConf) -> Result<()> {
             page_service::thread_main(conf, auth, pageserver_listener, conf.auth_type)
         })?;
 
-    page_service_thread
-        .join()
-        .expect("Page service thread has panicked")?;
+    join_handles.push(page_service_thread);
 
+    // Initialize tenant manager.
+    tenant_mgr::init(conf);
+
+    for handle in join_handles.into_iter() {
+        handle.join().unwrap().unwrap()
```

Comment:
That's 2 many unwraps :) Could you replace each `unwrap` with `expect(...)` with a corresponding description?